third_party/move/move-compiler-v2/src/bytecode_generator.rs

```diff
@@ -357,7 +368,7 @@ impl<'env> Generator<'env> {
                 self.scopes.pop();
             },
             ExpData::Mutate(id, lhs, rhs) => {
-                let rhs_temp = self.gen_arg(rhs, false);
+                let rhs_temp = self.gen_escape_auto_ref_arg(rhs, false);
```

Comment:
This is an interesting observation. Because of typing, the Mutate can never be in reference position, hence the escape doesn't make even sense. I removed the escape and added a comment.